crates/ty_python_semantic/src/types.rs

```diff
@@ -1655,6 +1655,20 @@ impl<'db> Type<'db> {
                 }
             }
 
+            (
+                Type::StringLiteral(_)
+                | Type::LiteralString
+                | Type::BooleanLiteral(_)
+                | Type::IntLiteral(_)
+                | Type::BytesLiteral(_)
+                | Type::ModuleLiteral(_),
+                _,
+            ) if (self.literal_fallback_instance(db))
+                .is_some_and(|instance| instance.is_assignable_to(db, target)) =>
```

Comment:
I think we can be a bit more general here -- this will make the logic more robust if we add literal-instance-fallbacks to any other types in the future

```suggestion
            _ if self.literal_fallback_instance(db))
                .is_some_and(|instance| instance.is_assignable_to(db, target) =>
```